KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java

```diff
@@ -181,6 +182,35 @@ private void loadAssetParams() {
 			impersonateClient = null;
 		}
 
+		public synchronized void setReadyForPlayback() {
+			if (readyForPlayback) {
+				logger.debug("Attention - READY_FOR_PLAYBACK already raised");
+				return;
+			}
+			if (!registered) {
+				logger.error("setReadyForPlayback called but stream is unregistered");
+				return ;
+			}
+			TimerTask setMediaServerTask = new TimerTask() {
+
+				@Override
+				public void run() {
+
+					if (registered) {
```

Comment:
if u got here, registered must be true, otherwise u would have exited the function